clang/lib/APINotes/APINotesManager.cpp

```diff
@@ -233,6 +234,10 @@ APINotesManager::getCurrentModuleAPINotes(Module *M, bool LookInModule,
 
         APINotes.push_back(*File);
       }
+      // If module FooCore is re-exported through module Foo, try Foo.apinotes.
+      if (!ExportedModuleName.empty())
+        if (auto File = findAPINotesFile(Dir, ExportedModuleName, WantPublic))
```

Comment:
I wonder what happens when the public module of a framework wants to be exported as the private module of a different framework...

Reply:
🤔 Yeah, would probably be good to have a diagnostic to reject that...